paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -281,7 +281,11 @@ internal class IntentConfirmationHandler(
                     PaymentConfirmationResult.Failed(
                         cause = nextStep.cause,
                         message = nextStep.message,
-                        type = PaymentConfirmationErrorType.Internal,
+                        type = if (nextStep.cause is InvalidDeferredIntentUsageException) {
+                            PaymentConfirmationErrorType.MerchantIntegration.InvalidDeferredIntentUsage
+                        } else {
+                            PaymentConfirmationErrorType.Internal
+                        },
```

Comment:
I could refactor `NextStep.Fail` to have an error type as well but sure if it's worth the effort given the small use case.

Reply:
I don't know anything about the code but I do wish it didn't take effort to log new types!